akka-http-caching/src/main/scala/akka/http/javadsl/server/directives/CachingDirectives.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.http.javadsl.server.directives
+
+import java.util.function.Supplier
+
+import akka.annotation.ApiMayChange
+import akka.http.caching.javadsl.Cache
+import akka.http.javadsl.model.Uri
+import akka.http.javadsl.server.{ RequestContext, Route, RouteResult }
+import akka.http.javadsl.model.HttpMethods.GET
+
+import akka.http.caching.CacheJavaMapping
+import akka.http.impl.util.JavaMapping
+
+import scala.concurrent.duration.Duration
+
+@ApiMayChange
+class CachingDirectives {
+
+  import akka.http.scaladsl.server.directives.{ CachingDirectives ⇒ D }
+
+  private implicit def routeResultCacheMapping[K] =
+    CacheJavaMapping.cacheMapping[K, RouteResult, K, akka.http.scaladsl.server.RouteResult]
+
+  /**
+   * Wraps its inner Route with caching support using the given [[akka.http.caching.scaladsl.Cache]] implementation and
+   * keyer function.
+   *
+   * Use [[akka.japi.JavaPartialFunction]] to build the `keyer`.
+   */
+  def cache[K](cache: Cache[K, RouteResult], keyer: PartialFunction[RequestContext, K], inner: Supplier[Route]) = RouteAdapter {
+    D.cache(
+      JavaMapping.toScala(cache),
+      toScalaKeyer(keyer)
+    ) { inner.get.delegate }
+  }
+
+  private def toScalaKeyer[K](keyer: PartialFunction[RequestContext, K]): PartialFunction[akka.http.scaladsl.server.RequestContext, K] = {
+    PartialFunction {
+      (scalaRequestContext: akka.http.scaladsl.server.RequestContext) ⇒
+        {
+          val javaRequestContext = akka.http.javadsl.server.RoutingJavaMapping.RequestContext.toJava(scalaRequestContext)
+          keyer(javaRequestContext)
+        }
+    }
+  }
+
+  /**
+   * A simple keyer function that will cache responses to *all* GET requests, with the URI as key.
+   * WARNING - consider whether you need special handling for e.g. authorised requests.
+   */
+  val simpleKeyer: PartialFunction[RequestContext, Uri] = {
```

Comment:
should this be private? or is this intended to be used by people?

Reply:
It's only used by the tests/example code. It could be moved out, but I'm ok with it remaining public as an example of a basic keyer (as it was in spray)

Reply:
I'd favor moving it out. It has all kinds of problems because it ignores important parts of a request, most importantly the existence of authentication information.